torch/distributed/fsdp/_exec_order_utils.py

```diff
@@ -173,10 +169,10 @@ def record_pre_forward(
         self._check_order(handle, is_training)
         # Fix the order after the first iteration and only record the first
         # usage of a handles key
-        if not self.is_first_iter or handle in self.handles_to_pre_forward_order_index:
+        if not self.is_first_iter or handle._pre_forward_order_index
```

Comment:
Haha fix syntax when you get the chance:
```suggestion
        if not self.is_first_iter or handle._pre_forward_order_index is not None:
```
(maybe we need to include the `is not None` explicitly instead of just `or handle._pre_forward_order_index` for the case it equals `0`?)